# Chapter 2 Exercise 4 from Think Stats, 2nd Edition




In [42]:
from __future__ import print_function, division

%matplotlib inline

import numpy as np

import nsfg
import first
import thinkstats2
import thinkplot

In [43]:
preg = nsfg.ReadFemPreg()
live = preg[preg.outcome == 1]

In [44]:
firsts = live[live.birthord == 1]
others = live[live.birthord != 1]

In [45]:
def CohenEffectSize(group1, group2):
    """Computes Cohen's effect size for two groups.
    
    group1: Series or DataFrame
    group2: Series or DataFrame
    
    returns: float if the arguments are Series;
             Series if the arguments are DataFrames
    """
    diff = group1.mean() - group2.mean()

    var1 = group1.var()
    var2 = group2.var()
    n1, n2 = len(group1), len(group2)

    pooled_var = (n1 * var1 + n2 * var2) / (n1 + n2)
    d = diff / np.sqrt(pooled_var)
    return d

## Exercises 4

Using the variable `totalwgt_lb`, investigate whether first babies are lighter or heavier than others. 

Compute Cohen’s effect size to quantify the difference between the groups.  How does it compare to the difference in pregnancy length?

In [46]:
# Using the given Cohen's effect size function on totalwgt_lb of firsts and others
wgt_firstvothers_effectsize = CohenEffectSize(firsts.totalwgt_lb, others.totalwgt_lb)

# Using the same function to investigate the effect size of pregnancy length
prglngth_firstvothers_effectsize = CohenEffectSize(firsts.prglngth, others.prglngth)

print(wgt_firstvothers_effectsize)
print(prglngth_firstvothers_effectsize)

-0.088672927072602
0.028879044654449883


Comparing the two effect sizes, it is clear that niether of the effects are significant. It is thought that any effect size less than 0.2 is considered small. However, some nuaces that can be drawn are that the effect size for weight is slightly larger than that of pregnancy length (|-0.089| - 0.029 = 0.060). Additionally, it should be noted that the effect size for weight is negative, indicating that first babies are lighter than other babies, whereas the effect size for pregnancy length is positive, indicating that first pregnancies are longer than other pregnancies. Still, these effect sizes are both too small to make any significant claims.